ballista/executor/src/circuit_breaker/stream.rs

```diff
@@ -55,7 +77,7 @@ impl Drop for CircuitBreakerStream {
 }
 
 pub trait CircuitBreakerCalculation {
-    fn calculate_delta(&mut self, batch: &RecordBatch) -> f64;
+    fn calculate_delta(&mut self, poll: &Poll<Option<Result<RecordBatch>>>) -> f64;
```

Comment:
hmm, maybe this should return `Option<f64>` so we don't send an update if there is none?

Reply:
If we don't send an update we might not receive the trip signal in time, right?
So it should be sent as often as possible, right?

Reply:
The trip signal is not coupled to the send updates I thought?

Reply:
Currently it is, when we update the received value in the controller we determine if it's tripped, then respond back with the trip signal (inside a batch of requests and responses, though).

Reply:
That seems like something we want to fix. Since the client will send updates in batches (with a timeout) maybe we can:
1. Send the update even if we time out and there are not updates
2. Have the response include trip signals for anything that has been tripped on that executor

That way we get an implicit "heartbeat" for trip signals. In the future we can probably unify this with the normal executor heartbeat to eliminate the extra rpc.